honeybee_energy/simulation/output.py

```diff
@@ -0,0 +1,518 @@
+                An empty list or None will result in no summary reports.
+                Default: ['AllSummary']. See the Input Output Reference SummaryReports
+                section for a full list of all reports that can be requested.
+                (https://bigladdersoftware.com/epx/docs/9-1/input-output-reference/
+                output-table-summaryreports.html#outputtablesummaryreports).
+        """
+        self.outputs = outputs
+        self.reporting_frequency = reporting_frequency
+        self.include_sqlite = include_sqlite
+        self.include_html = include_html
+        self.summary_reports = summary_reports
+
+    @property
+    def outputs(self):
+        """Get or set a tuple of EnergyPlus output names as strings.
+
+        These outputs will be requested from the simulation and, if None,
+        no outputs will be requested.
+        """
+        return tuple(self._outputs)
+
+    @outputs.setter
+    def outputs(self, value):
+        if value is not None:
+            if not isinstance(value, list):
+                value = list(value)
+            for output in value:
+                assert isinstance(output, str), 'Output {} is not valid'.format(output)
+        else:
+            value = []
+        self._outputs = value
+
+    @property
+    def reporting_frequency(self):
+        """Get or set text for the frequency at which the outputs are reported.
+
+        Choose from the following:
+            * Annual
+            * Monthly
+            * Daily
+            * Hourly
+            * Timestep
+        """
+        return self._reporting_frequency
+
+    @reporting_frequency.setter
+    def reporting_frequency(self, value):
+        value = value.title()
+        assert value in self.REPORTING_FREQUENCIES, 'reporting_frequency {} ' \
+            'is not recognized.\nChoose from the following:\n{}'.format(
+                value, self.REPORTING_FREQUENCIES)
+        self._reporting_frequency = value
+
+    @property
+    def include_sqlite(self):
+        """Get or set a boolean for whether a SQLite report should be generated."""
+        return self._include_sqlite
+
+    @include_sqlite.setter
+    def include_sqlite(self, value):
+        self._include_sqlite = bool(value)
+
+    @property
+    def include_html(self):
+        """Get or set a boolean for whether an HTML report should be generated."""
+        return self._include_html
+
+    @include_html.setter
+    def include_html(self, value):
+        self._include_html = bool(value)
+
+    @property
+    def summary_reports(self):
+        """Get or set a tuple of EnergyPlus summary report names as strings.
+
+        These reports will be requested from the simulation and, if None,
+        no summary reports will be written.
+        """
+        return tuple(self._summary_reports)
+
+    @summary_reports.setter
+    def summary_reports(self, value):
+        if value is not None:
+            if not isinstance(value, list):
```

Comment:
This will fail for tuple or iterator inputs. A better test is probably checking against being string or bytes.

Reply:
I don't know what you mean and you might be misunderstanding what this code is doing. If someone inputs a tuple or an iterator, I want it to be cast to a list. It's not going to fail in this case.

Reply:
Sorry. You are right. I misread the code. But now this code fails if one inputs a string. This check will create a list of characters.

Reply:
Ah, that's a good point. I always forget strings are also iterables and, particularly in this case, it will result in a very common error where someone passes a string of a single E+ output. Ok, I'll add an extra check in here.